godotproject/DeathPlane.gd

```diff
@@ -1,22 +1,6 @@
 extends Area2D
 
-
-# Declare member variables here. Examples:
-# var a = 2
-# var b = "text"
-
-
-# Called when the node enters the scene tree for the first time.
-func _ready():
-	pass # Replace with function body.
-
-
-# Called every frame. 'delta' is the elapsed time since the previous frame.
-#func _process(delta):
-#	pass
-
-
 func _on_DeathPlane_body_entered(body):
-	if body.has_method("die"):
-		body.die()
-	pass # Replace with function body.
+	if  body.name == "Player":
+		body.takeDamage(1);
+		Global.fade_to_scene("res://Level" + str(int(get_tree().current_scene.name)) + ".tscn")
```

Comment:
hmm thats a good idea. I was thinking we could have a a few different death planes that all reset the player to a different position further along the level, kinda acting as checkpoints. I'll look into setting the players position, but it would still be nice to have the fade.